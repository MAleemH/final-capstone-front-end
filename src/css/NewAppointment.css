* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: 'Montserrat', sans-serif;
    letter-spacing: 0.2px;
  }
  
  .list_style_none {
    list-style: none;
  }
  
  .td_none {
    text-decoration: none;
  }
  
  .fieldset_border_none {
    border: none;
  }
  
  .new_therapist_body {
    background-color: #fffcf6;
  }
  
  .new_therapist_header {
    width: 90%;
    margin: auto;
    padding: 10px 0;
    height: 40px;
    position: sticky;
    top: 0;
    left: 0;
    background: transparent;
  }
  
  .new_therapist_header img {
    height: 30px;
  }
  
  .new_therapist_main {
    width: 40%;
    margin: auto;
  }
  
  .logo_fig {
    display: flex;
    justify-content: center;
    padding: 65px 0;
  }
  
  .eyebox img {
    width: 30px;
  }
  
  .image_input_container img {
    width: 40px;
  }
  
  .new_therapist_main .logo_fig img {
    height: 100px;
    margin: auto;
  }
  
  .fadebreak_box {
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
  }
  
  .new_therapist_form {
    padding: 20px 0;
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    gap: 15px;
  }
  
  .new_therapist_form fieldset {
    display: flex;
    flex-direction: column;
    gap: 15px;
    width: 100%;
  }
  
  .new_therapist_form .input_name {
    width: 100%;
    color: #172b4d;
    font-size: 15px;
    padding: 15px;
    outline: none;
    border: 1px solid #d9e1f6;
    border-radius: 10px;
  }
  
  .password_box {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #fff;
    border: 1px solid #d9e1f6;
    border-radius: 10px;
    padding-right: 5px;
  }
  
  .new_therapist_form .input_password {
    width: 100%;
    color: #172b4d;
    font-size: 15px;
    padding: 15px;
    outline: none;
    border: none;
    border-radius: 10px;
  }
  
  .new_therapist_form input::placeholder {
    color: #172b4d;
    font-size: 17px;
    font-weight: 400;
    line-height: 28px;
    transition: 3s;
  }
  
  .new_therapist_form input:hover::placeholder {
    transform: translate(10px, 0);
    color: #172b4d50;
  }
  
  .new_therapist_action {
    width: 20%;
    margin: auto;
  }
  
  .new_therapist_action button {
    padding: 10px 8px;
    border-radius: 10px;
    background-color: #ff9186;
    border: 1px solid #d9e1f6;
    color: #172b4d;
  }
  
  .new_therapist_main .new_therapist_main_div {
    display: flex;
    justify-content: end;
    gap: 20px;
    align-items: center;
  }
  
  .new_therapist_footer ul a {
    color: #172b4d;
  }
  
  .new_therapist_main .new_therapist_main_div a {
    color: #172b4d;
  }
  
  .new_therapist_main .new_therapist_main_div button {
    color: #172b4d;
    padding: 10px 8px;
    border-radius: 10px;
    background-color: #ff9186;
    border: 1px solid #d9e1f6;
  }
  
  .new_therapist_footer {
    width: 90%;
    margin: auto;
    position: sticky;
    bottom: 0;
    left: 0;
    background: transparent;
    padding: 10px 0;
  }
  
  .new_therapist_footer ul {
    display: flex;
    gap: 15px;
  }
  
  .password_display {
    border: none;
    background-color: transparent;
  }
  
  .rotate_breakline {
    transform: rotate(180deg);
  }
  
  .image_input_container {
    position: relative;
    display: flex;
    gap: 15px;
    align-items: center;
    background-color: white;
  }
  
  .image_input_container input[type="file"] {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    opacity: 0;
    cursor: pointer;
  }
  
  .image_input_container p {
    cursor: pointer;
    color: #172b4d;
  }
  
  select {
    background-color: #fffcf6;
  }
  
  option {
    background-color: #fffcf6;
  }
  