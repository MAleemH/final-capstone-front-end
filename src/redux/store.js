import { configureStore } from '@reduxjs/toolkit';
import therapyReducer from './Therapy/therapySlice';
import userReducer from './User/userSlice';

const store = configureStore({
  reducer: {
    therapy: therapyReducer,
    user: userReducer,
  },
});

export default store;
