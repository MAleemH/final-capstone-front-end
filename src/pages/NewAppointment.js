import React, { useEffect, useState } from 'react';
import { Link, useNavigate } from 'react-router-dom';
// import { useDispatch } from 'react-redux';
// import { postReserve } from '../redux/Reserve/reserveSlice';
import '../css/NewAppointment.css';
import specializationArr from '../components/specialization';
import backImg from '../img/back.png';
import fadingBreak from '../img/fading_break.png';
import logoImg from '../img/logo.png';

function NewAppointment() {
  // const dispatch = useDispatch();
  const [address, setAddress] = useState('');
  const [timestampp, setTimestampp] = useState(0);
  const [specialty, setSpecialty] = useState('');
  const [therapist, setTherapist] = useState({});
  const [errorMessage, setErrorMessage] = useState('');
  const navigate = useNavigate();

  const goBack = () => {
    navigate(-1);
  };

  const nullReserveData = async () => {
    setAddress('');
    setTimestampp('');
    setSpecialty('');
    setTherapist({});
  };

  const handleNewAppointment = async () => {
    const timestamp = (new Date(timestampp)).toISOString().substr(0, 16);
    console.log(address, timestamp, therapist);
    const userData = {
      address, date: timestamp, therapist,
    };
    console.log(userData);
    // dispatch(postReserve(userData))
    await nullReserveData();
    navigate('appointments');
  };

  useEffect(() => {
    let active = true;
    (async () => {
      if (active) {
        setErrorMessage('');
      }
    })();
    return () => {
      active = false;
    };
  }, []);

  return (
    <div className="new_appointment_body">

      <header className="new_appointment_header">
        <nav>
          <button className="back_none" type="button" onClick={goBack}>
            <img src={backImg} alt="" />
          </button>
        </nav>
      </header>

      <main className="new_appointment_main">
        <figure className="logo_fig">
          <Link to="/">
            <img src={logoImg} alt="" />
          </Link>
        </figure>

        <div className="fadebreak_box">
          <figure>
            {' '}
            <img src={fadingBreak} alt="" />
            {' '}
          </figure>
          <p>
            &nbsp;
            {`${errorMessage || 'Reserve'}`}
              &nbsp;
          </p>
          <figure>
            {' '}
            <img className="rotate_breakline" src={fadingBreak} alt="" />
            {' '}
          </figure>
        </div>

        <form action="" className="new_appointment_form">
          <fieldset className="fieldset_border_none">

            <input className="input_name" type="text" placeholder="Address (State and City)" aria-label="Input Address" value={address} onChange={(e) => setAddress(e.target.value)} required />

            <input className="input_name" type="datetime-local" placeholder="Time" aria-label="Input Date and Time" value={timestampp.name} onChange={(e) => setTimestampp(e.target.value)} required />

            <select aria-label="Input Label" className="input_name" id="specializationId" value={specialty.name} onChange={(e) => setSpecialty(e.target.value)}>
              {specializationArr.map((specialty) => (
                <option key={specialty.id} value={specialty.value} aria-label="Input Specialization">{specialty.name}</option>
              ))}
            </select>

            <select aria-label="Input Label" className="input_name" id="therapistId" value={therapist.name} onChange={(e) => setTherapist(e.target.value)}>
              {/* Filter based on specialty choosen to display doctors in the specialty */}
              <option value="" aria-label="Input Therapist">Therapist Name</option>
              {/* ))} */}
            </select>

          </fieldset>

          <fieldset className="fieldset_border_none new_appointment_action">
            <button type="button" onClick={handleNewAppointment}> Book</button>
          </fieldset>
        </form>

      </main>

      <footer className="new_appointment_footer">
        <ul className="list_style_none">
          <li><Link to="/new_appointment" className="td_none"> terms </Link></li>
          <li><Link to="/new_appointment" className="td_none">  privacy </Link></li>
        </ul>
      </footer>

    </div>
  );
}

export default NewAppointment;
