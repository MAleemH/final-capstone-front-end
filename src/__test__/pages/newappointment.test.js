import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import NewAppointment from '../../pages/NewAppointment';
import specializationArr from '../../components/speciliazation';

it('NewAppointment component renders correctly', () => {
  const myTherapists = store.getState().therapy.therapists;
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <NewAppointment myTherapists={myTherapists} specializationArr={specializationArr} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
