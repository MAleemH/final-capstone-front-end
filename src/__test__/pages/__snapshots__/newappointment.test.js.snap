// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewAppointment component renders correctly 1`] = `
<div
  className="new_appointment_body"
>
  <header
    className="new_appointment_header"
  >
    <nav>
      <button
        className="back_none"
        onClick={[Function]}
        type="button"
      >
        <img
          alt=""
          src="back.png"
        />
      </button>
    </nav>
  </header>
  <main
    className="new_appointment_main"
  >
    <figure
      className="logo_fig"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          src="logo.png"
        />
      </a>
    </figure>
    <div
      className="fadebreak_box"
    >
      <figure>
         
        <img
          alt=""
          src="fading_break.png"
        />
         
      </figure>
      <p>
         
        Reserve
         
      </p>
      <figure>
         
        <img
          alt=""
          className="rotate_breakline"
          src="fading_break.png"
        />
         
      </figure>
    </div>
    <form
      action=""
      className="new_appointment_form"
    >
      <fieldset
        className="fieldset_border_none"
      >
        <input
          aria-label="Input Address"
          className="input_name"
          onChange={[Function]}
          placeholder="Address (State and City)"
          required={true}
          type="text"
          value=""
        />
        <input
          aria-label="Input Date and Time"
          className="input_name"
          onChange={[Function]}
          placeholder="Time"
          required={true}
          type="datetime-local"
        />
        <select
          aria-label="Input Label"
          className="input_name"
          id="specializationId"
          onChange={[Function]}
        >
          <option
            aria-label="Input Specialization"
            value=""
          >
            Select Specialization
          </option>
          <option
            aria-label="Input Specialization"
            value="Cardiovascular"
          >
            Cardiovascular
          </option>
          <option
            aria-label="Input Specialization"
            value="Emotional"
          >
            Emotional
          </option>
          <option
            aria-label="Input Specialization"
            value="Romance"
          >
            Romance
          </option>
          <option
            aria-label="Input Specialization"
            value="Sexual"
          >
            Sexual
          </option>
          <option
            aria-label="Input Specialization"
            value="Academic"
          >
            Academic
          </option>
          <option
            aria-label="Input Specialization"
            value="Career"
          >
            Career
          </option>
          <option
            aria-label="Input Specialization"
            value="Marital"
          >
            Marital
          </option>
          <option
            aria-label="Input Specialization"
            value="Dating"
          >
            Dating
          </option>
        </select>
        <select
          aria-label="Input Label"
          className="input_name"
          disabled={true}
          id=""
          onChange={[Function]}
          value={0}
        >
          <option
            value=""
          >
            Select a therapist
          </option>
        </select>
      </fieldset>
      <fieldset
        className="fieldset_border_none new_appointment_action"
      >
        <button
          onClick={[Function]}
          type="button"
        >
           Book
        </button>
      </fieldset>
    </form>
  </main>
  <footer
    className="new_appointment_footer"
  >
    <ul
      className="list_style_none"
    >
      <li>
        <a
          className="td_none"
          href="/new_appointment"
          onClick={[Function]}
        >
           terms 
        </a>
      </li>
      <li>
        <a
          className="td_none"
          href="/new_appointment"
          onClick={[Function]}
        >
            privacy 
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
